File: 백준/Gold/1069. 집으로/집으로.py
X, Y, D, T = map(int, input().split(" "))

dist = (X**2 + Y**2)**0.5

# jump = dist 방향으로 점프할 수 있는 최대 경우
jump = dist//D

if dist >= D:
    case1 = T * jump + (dist - (D * jump))
    case2 = dist
    case3 = T * (jump + 1)
    
    result = min(case1, min(case2, case3))
    result = round(result,9)
    print(result)

else:
    case1 = T + (D - dist)
    case2 = dist
    case3 = T * 2
    
    result = min(case1, min(case2, case3))
    result = round(result,9)
    print(result)

File: 백준/Gold/1069. 집으로/집으로.cc
#include <stdio.h>
#include <cmath>
#define _CRT_SECURE_NO_WARNINGS

double distance(double y, double x) {
	return sqrt(pow(y, 2) + pow(x, 2));
}

int main() {
	double x, y, d, t;
	scanf("%lf %lf %lf %lf", &x, &y, &d, &t);

	double dis = distance(y, x);
	double case1, case2, case3;
	case1 = dis;
	case2 = dis >= d ? floor(dis / d) * t + fmod(dis, d) : t + d - dis;
	case3 = dis >= d ? (floor(dis / d) + 1) * t : t * 2;
	
	printf("%.9lf", fmin(case1, fmin(case2, case3)));
	return 0;
}